src/hooks/useDomainAvailability.ts

```diff
@@ -0,0 +1,64 @@
+import { useQuery } from 'react-query';
+import { useBuyDomain } from './useBuyDomain';
+import { useZnsSdk } from './useZnsSdk';
+import { MIN_DOMAIN_NAME_LENGTH } from '../constants/domains';
+
+export interface UseDomainAvailabilityReturn {
+	isLoading: boolean;
+	isDomainAvailable: boolean;
+	isDomainSearchEnabled: boolean;
+	domainPrice?: string;
+}
+
+export const useDomainAvailability = (): UseDomainAvailabilityReturn => {
+	const sdk = useZnsSdk();
+	const { domainName } = useBuyDomain();
+
+	const isDomainSearchEnabled = domainName.length >= MIN_DOMAIN_NAME_LENGTH;
+
+	// Query
+	const { isLoading, data: { isDomainAvailable, domainPrice } = {} } = useQuery(
```

Comment:
As we discussed in Discord - can we return the whole `useQuery` to cut down on boilerplate? Happy to discuss this if you disagree.